File: Blind75/01_Arrays_and_Hashing/02_valid_anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] arr = new int[26];

        if (s.length() != t.length()) {
            return false;
        }

        for (int i = 0; i < s.length(); i++) {
            arr[s.toCharArray()[i] - 'a'] += 1;
            arr[t.toCharArray()[i] - 'a'] -= 1;
        }

        for (int x : arr) {
            if (x != 0)
                return false;
        }

        return true;
    }
}

File: Blind75/01_Arrays_and_Hashing/02_valid_anagram.cpp
class Solution
{
public:
    bool isAnagram(string s, string t)
    {

        vector<int> count(26);

        int m = s.length(), n = t.length();

        if (m != n)
            return false;

        for (int i = 0; i < n; i++)
        {
            count[s[i] - 'a']++;
            count[t[i] - 'a']--;
        }

        for (int x : count)
            if (x != 0)
                return false;
        return true;
    }
};